cmd/kube-apiserver/app/server.go

```diff
@@ -569,7 +575,51 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 	}
 
 	if s.Etcd.EncryptionProviderConfigFilepath != "" {
-		transformerOverrides, err := encryptionconfig.GetTransformerOverrides(s.Etcd.EncryptionProviderConfigFilepath)
+		// Obtain GCE cloud for a Google Cloud KMS connection if needed.
+		// cloudproviders package is in the main kubernetes directory, and thus we cannot access
+		// the gce package (sub-package of cloudproviders) from inside the staging directory.
+		// We need to be able to create a gce cloud object inside the staging directory. Thus, for now,
+		// we pass a function which does that, from here.
+		// If cloudproviders package is moved to a new location, move the logic of this function to kms.Factory.
+		getGCECloud := func() (*kms.GCECloud, error) {
+			cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider.CloudProvider, s.CloudProvider.CloudConfigFile)
+			if err != nil {
+				return nil, err
+			}
+
+			var cloudkmsService *cloudkms.Service
+			var projectID string
+
+			// This check is false if cloud is nil, or is not an instance of gce.GCECloud.
+			if gke, ok := cloud.(*gce.GCECloud); ok {
```

Comment:
Is all this GCE-specific code necessary in the apiserver? Can it be abstracted away via Cloud interface? (I know nothing about this feature)